src/core/grid.js

```diff
@@ -156,6 +158,52 @@ class Grid {
       this.markAsMoved()
     }
   }
+
+  checkReached2048(value) {
+    if (value === 2048) this.reached2048 = true;
+  }
+
+  canMove() {
+    for (let row = 0; row < this.size; row++) {
+      for (let col = 0; col < this.size; col++) {
+        const currentCell = this.cells[row][col];
+
+        if (currentCell.isEmpty()) return true;
+
+        if (this.canMergeRight(row, col, currentCell) ||
+          this.canMergeDown(row, col, currentCell)) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  canMergeRight(row, col, currentCell) {
+    if (col < LAST_INDEX) {
+      const rightCell = this.cells[row][col + 1];
+      return rightCell.tile && this.isMergePossible(currentCell.tile, rightCell.tile);
```

Comment:
Make the return simpler
```suggestion
      const mergePossible = this.isMergePossible(currentCell.tile, rightCell.tile);
      return rightCell.tile && mergePossible;
```